### ISSUE TITLE
[BITV]: Focus trap on search field doesn't work well

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#39434]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

This is accessibility issue.

Focus trap on search field "Search for share recipients" doesn't work well.

### Steps to reproduce

**Bug is only reproducible with opened photo viewer.**. If viewer is closed, all focus trap and keyboard navigation working well.

1. Open any picture in files viewer
2. Open sidebar
3. Click to "sharing" tab
4. Focus on the search field "Search for share recipients" 
5. Try to exit search field: Viewer with photo will be closed
6. Try to tab forwards: that doesn't work. Only tab backwards works.

![Peek 2023-07-17 12-04](https://github.com/nextcloud/server/assets/6078378/1bc2c442-c00a-4c82-b33e-a9797b9146a2)

### Expected behavior

1. Open any picture in files viewer
2. Open sidebar
3. Click to "sharing" tab
4. Focus on the search field "Search for share recipients" 
5. Try to exit search field: Viewer with photo won't be closed. Search field will be closed, but still focused.
6. Try to tab forwards: tabulation works. Next focused element would be the next element in a sidebar below search field.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by szaimen** [comment]
Cc @nextcloud/server-frontend

**Comment by rheindevde** [comment]
Exact same issue for my installation. Latest AIO-deployment, no encryption, default user backend, no fancy stuff going on.
I'm sharing via link to temporarily work around this.
